### ISSUE TITLE
Mods can't add new mixin configs using `Mixins.addConfiguration`

### ISSUE THREAD

**Issue by melontini** [Sinytra/Connector#557]
### Describe the bug

Same as title.

### Steps to reproduce

Steps to reproduce:
1. Run this dev version of Andromeda https://github.com/melontini/andromeda/actions/runs/7021209448.
2. Notice that the mixins have not been applied.

### Logs

Silent failures, even in debug.log

### Additional context

The way Andromeda adds dynamic configs is pretty unsafe, but even `Mixins.addConfiguration("andromeda.mixins.json)"` doesn't work.

**Comment by Su5eD** [comment]
Yeah, this is gonna be tricky. Mixin never loads andromeda's modular configs despite "queueing" them because they're only added after mixin had already transformed a class. This check exists so that every mixin has a chance of modifying its target class when it gets loaded, as modifications are not possible afterwards.

And here comes the launch order issue again! Mods are placed on the `GAME` layer, which is transformable. Under ideal circumstances, I would run the `preLaunch` entrypoint just before the main method is invoked. However, due to modlauncher only loading launch plugins from the BOOT layer, other mods tend to inject their own ones using reflection. There's one mod in particular that I found to be incompatible with this approach: REIPC.

REIPC uses a mixin plugin as an entrypoint to inject its launch plugin. This works fine under normal circumstances. However, if I run preLaunch inside `ILaunchPluginService#initializeLaunch` (which runs inside a for loop of the map that REIPC injects plugins into), it results in a ConcurrentModificationException. In an effort to avoid this situation, Connector adjusts its launch sequence to run preLaunch only after the plugin iteration is over.

Another issue is Forge's modular layout. The `GAME` layer is transformable, while also containing preLaunch and mixin plugin code. It is therefore impossible to invoke preLaunch without initializing mixin plugins in the process.  
I'll have a closer look at the issue and see what our options are.

**Comment by melontini** [comment]
Oh, thanks for the detailed explanation!
Have you looked into `IExtension.checkActive`? It seems pretty close to the preLaunch entrypoint. MixinExtras also uses `IExtension`, but `preApply`, which might be late.

**Comment by melontini** [comment]
So, either this is an unsolvable puzzle, or I'm missing some important context.

If I understand correctly, Fabric's load order, for mods, is:

1. Fabric loads language adapters.
2. After classpath is unlocked, on first classload, Mixin selects the environment and loads configs.
3. PreLaunch is executed.
4. GameProvider launches.

The immediate window trick should work correctly, right? No game/mod classes are loaded before that point (except Mixin plugins and whatever loads with them, but they aren't transformable by mixins). If Mixin counted every processed class as transformed, addConfiguration wouldn't work on Fabric.

> mixin had already transformed a class

But which one? Why would this be the case?

Anyways, there's no other place for the entrypoint outside the loop. It's possible to wrap the ILaunchHandlerService, but this doesn't give us much, as it's before Mixin loads plugins.
